src/Microsoft.AspNetCore.ResponseCaching/Internal/ResponseCacheEntrySerializer.cs

```diff
@@ -2,16 +2,18 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.ResponseCaching.Internal
 {
-    internal static class CacheEntrySerializer
+    internal static class ResponseCacheEntrySerializer
     {
+        private static readonly List<byte[]> EmptyList = new List<byte[]>();
         private const int FormatVersion = 1;
 
-        public static object Deserialize(byte[] serializedEntry)
+        public static IResponseCacheEntry Deserialize(byte[] serializedEntry)
         {
             if (serializedEntry == null)
```

Comment:
Of course, but that would be an invalid entry and the exception will be handled by the DistributedResponseCacheStore.